### ISSUE TITLE
onnx export error:

### ISSUE THREAD

**Issue by xinsuinizhuan** [ultralytics/yolov5#6112]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

C:\ProgramData\Anaconda3\envs\yolov5\python.exe E:/Item/Item_done/yolo/yolo5/yolov5_v6.0/yolov5/export.py
export: data=data\coco128.yaml, weights=yolov5s6.pt, imgsz=[640, 640], batch_size=1, device=cpu, half=False, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['torchscript', 'onnx']
YOLOv5  v6.0-114-ga4207a2 torch 1.7.1+cu110 CPU

Fusing layers... 
Model Summary: 280 layers, 12612508 parameters, 0 gradients

PyTorch: starting from yolov5s6.pt (25.7 MB)

TorchScript: starting export with torch 1.7.1+cu110...
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:
TorchScript: export success, saved as yolov5s6.torchscript (51.0 MB)

ONNX: starting export with onnx 1.10.2...
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:

Process finished with exit code -1073741819 (0xC0000005)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by xinsuinizhuan** [comment]
(yolov5) E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5>pip list
Package                 Version
----------------------- ---------------------
absl-py                 1.0.0
addict                  2.4.0
cachetools              4.2.4
certifi                 2021.10.8
charset-normalizer      2.0.9
click                   8.0.3
colorama                0.4.4
cycler                  0.11.0
defusedxml              0.7.1
editdistance            0.6.0
fast_ctc_decode         0.3.0
flatbuffers             2.0
fonttools               4.28.5
future                  0.18.2
google-auth             2.3.3
google-auth-oauthlib    0.4.6
grpcio                  1.43.0
hyperopt                0.1.2
idna                    3.3
imageio                 2.13.5
importlib-metadata      4.10.0
joblib                  1.1.0
jstyleson               0.0.2
kiwisolver              1.3.2
Markdown                3.3.6
matplotlib              3.5.1
networkx                2.6.3
nibabel                 3.2.1
nltk                    3.6.6
numpy                   1.19.5
oauthlib                3.1.1
onnx                    1.10.2
onnx-simplifier         0.3.6
onnxoptimizer           0.2.6
onnxruntime             1.10.0
onnxruntime-gpu         1.10.0
opencv-contrib-python   4.5.4+dummy
opencv-python           4.5.4.60
openvino                2021.4.2
openvino-dev            2021.4.2
packaging               21.3
pandas                  1.1.5
parasail                1.2.4
Pillow                  8.4.0
pip                     21.3.1
progress                1.6
protobuf                3.19.1
py-cpuinfo              8.0.0
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pydicom                 2.2.2
pymongo                 4.0.1
pyparsing               3.0.6
python-dateutil         2.8.2
pytz                    2021.3
PyWavelets              1.2.0
PyYAML                  6.0
rawpy                   0.17.0
regex                   2021.11.10
requests                2.26.0
requests-oauthlib       1.3.0
rsa                     4.8
scikit-image            0.19.1
scikit-learn            1.0.2
scipy                   1.5.4
seaborn                 0.11.2
sentencepiece           0.1.96
setuptools              60.1.0
Shapely                 1.8.0
six                     1.16.0
tensorboard             2.7.0
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.0
texttable               1.6.4
thop                    0.0.31.post2005241907
threadpoolctl           3.0.0
tifffile                2021.11.2
tokenizers              0.11.0
torch                   1.7.1+cu110
torchaudio              0.7.2
torchvision             0.8.2+cu110
tqdm                    4.62.3
typing_extensions       4.0.1
urllib3                 1.26.7
Werkzeug                2.0.2
wheel                   0.37.1
yamlloader              1.1.0
zipp                    3.6.0

**Comment by glenn-jocher** [comment]
@xinsuinizhuan it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual Python 3.8 environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install -r requirements.txt` again. We also highly recommend using one of our verified environments below. 


## Requirements

[**Python>=3.6.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
$ git clone https://github.com/ultralytics/yolov5
$ cd yolov5
$ pip install -r requirements.txt
```

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by xinsuinizhuan** [comment]
python version is 3.8, is also no work:
NNX: starting export with onnx 1.10.2...
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:

Process finished with exit code -1073741819 (0xC0000005)


my cuda version is 11.0
my setep is:

windows anaconda:

conda create -n yolov5 python=3.8
conda activate yolov5
cd ... yolov5 path,then:
pip install -r requirements.txt
pip unisntall torch
pip unisntall torchsivion
pip unisntall onnxruntime
pip install torch==1.7.1+cu110 torchvision==0.8.2+cu110 torchaudio==0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
pip install onnxruntime-gpu

becouse i want to train ,so  must torch gpu version, and i want test onnxruntime on gpu ,so must onnxruntine-gpu.

how should i do?

**Comment by glenn-jocher** [comment]
@xinsuinizhuan if your local environment is not working then just export in a working environment like Colab, as the above response already recommends.

**Comment by xinsuinizhuan** [comment]
OK。Thank you. But it works befor o month, i only update it these days, then it not works.

**Comment by xinsuinizhuan** [comment]
> @xinsuinizhuan if your local environment is not working then just export in a working environment like Colab, as the above response already recommends.

C:\ProgramData\Anaconda3\envs\yolov5\python.exe E:/Item/Item_done/yolo/yolo5/yolov5_v6.0/yolov5/export.py
export: data=data\coco128.yaml, weights=yolov5s.pt, imgsz=[640, 640], batch_size=1, device=cpu, half=False, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['torchscript', 'onnx']
YOLOv5  2021-12-23 torch 1.10.1+cpu CPU

Fusing layers... 
Model Summary: 213 layers, 7225885 parameters, 0 gradients

PyTorch: starting from yolov5s.pt (14.7 MB)

TorchScript: starting export with torch 1.10.1+cpu...
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:
TorchScript: export success, saved as yolov5s.torchscript (29.4 MB)

ONNX: starting export with onnx 1.10.2...
E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5\models\yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.onnx_dynamic or self.grid[i].shape[2:4] != x[i].shape[2:4]:

Process finished with exit code -1073741819 (0xC0000005)


I try it, but it also not work.
setep:

windows anaconda:
conda create -n yolov5 python=3.8
conda activate yolov5
cd ... yolov5 path,then:
pip install -r requirements.txt

then export.py

**Comment by xinsuinizhuan** [comment]
(yolov5) E:\Item\Item_done\yolo\yolo5\yolov5_v6.0\yolov5>pip list
Package                 Version
----------------------- ---------------------
absl-py                 1.0.0
cachetools              4.2.4
certifi                 2021.10.8
charset-normalizer      2.0.9
colorama                0.4.4
cycler                  0.11.0
flatbuffers             2.0
fonttools               4.28.5
google-auth             2.3.3
google-auth-oauthlib    0.4.6
grpcio                  1.43.0
idna                    3.3
importlib-metadata      4.10.0
kiwisolver              1.3.2
Markdown                3.3.6
matplotlib              3.5.1
numpy                   1.21.5
oauthlib                3.1.1
onnx                    1.10.2
onnx-simplifier         0.3.6
onnxoptimizer           0.2.6
onnxruntime             1.10.0
opencv-python           4.5.4.60
packaging               21.3
pandas                  1.3.5
Pillow                  8.4.0
pip                     21.3.1
protobuf                3.19.1
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pyparsing               3.0.6
python-dateutil         2.8.2
pytz                    2021.3
PyYAML                  6.0
requests                2.26.0
requests-oauthlib       1.3.0
rsa                     4.8
scipy                   1.7.3
seaborn                 0.11.2
setuptools              60.1.0
six                     1.16.0
tensorboard             2.7.0
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.0
thop                    0.0.31.post2005241907
torch                   1.10.1
torchvision             0.11.2
tqdm                    4.62.3
typing_extensions       4.0.1
urllib3                 1.26.7
Werkzeug                2.0.2
wheel                   0.37.1
zipp                    3.6.0

**Comment by xinsuinizhuan** [comment]
sorry, i change my pycharm version, it works.

**Comment by HuKai97** [comment]
i run export.py in cmd，it works too.....

**Comment by xinsuinizhuan** [comment]
yes. It is my error of the pycharm version. Change it, it  works.

**Comment by VShawn** [comment]
this ticket reproduction on my vscode environment.

![image](https://user-images.githubusercontent.com/10143738/149086692-12c6d4cf-b52e-4241-a37b-1e9f9a5fce02.png)

![image](https://user-images.githubusercontent.com/10143738/149086850-e9026d27-6c32-4f0c-8db7-ab61fc85bfbc.png)

**Comment by VShawn** [comment]
also reproduction on CoLab

![image](https://user-images.githubusercontent.com/10143738/149089924-3e9420c5-1964-47bf-9565-df9557e3fa2e.png)

here is the book:

https://colab.research.google.com/drive/1arIrnfdB2GcUEa9YZuiFfxbGE03GbxtS?usp=sharing


---
BTW can you share me a exported `yolov5s6` onnx file for a benchmark test?

**Comment by glenn-jocher** [comment]
@VShawn this is a warning, not an error.

**Comment by VShawn** [comment]
@glenn-jocher 

Yes it's just a warning.

but since the .onnx grows to 52.1 MB (while the .pt is 25.7MB), the onnx model slower than the pytorch model on yolov5s6. I think this may relate to the warning.

## yolov5s6 time usage on pytorch and onnx

```
pytorch torch.Size([1, 3, 1280, 1280])  time avg-cost = 0.01874645233154297 s ( 53.343426388863456 fps)
onnx-runtime [1, 3, 1280, 1280]  time avg-cost = 0.04908282995223999 s ( 20.37372337685193 fps)
```

## As a comparison:  mobilenetv3 time usage  on pytorch and onnx

```
pytorch torch.Size([1, 3, 224, 224])  time avg-cost = 0.011908938884735107 s ( 83.97053756668457 fps)
onnx-runtime torch.Size([1, 3, 224, 224])  time avg-cost = 0.0036801528930664063 s ( 271.7278409503177 fps)
```

**Comment by glenn-jocher** [comment]
@VShawn filesize is a completely different topic. PyTorch models are saved in FP16 and ONNX models export by default at FP32.

**Comment by glenn-jocher** [comment]
ONNX FP32 GPU - 17 ms
<img width="1077" alt="Screen Shot 2022-01-12 at 2 20 13 PM" src="https://user-images.githubusercontent.com/26833433/149243684-fad96d36-510d-4bfc-95ec-3a3bb9114070.png">


ONNX FP16 GPU - 12 ms
<img width="1074" alt="Screen Shot 2022-01-12 at 2 20 55 PM" src="https://user-images.githubusercontent.com/26833433/149243767-b6b7aa7b-0eb7-49cc-89c2-9b37da48a66a.png">

**Comment by VShawn** [comment]
> PyTorch models are saved in FP16 and ONNX models export by default at FP32.

I didn't know it! I'm new on onnx.

> the onnx model slower than the pytorch model on yolov5s6. I think this may relate to the warning.

Then I was wrong, I will go and check why onnx is 2x slower then pytorch.

![image](https://user-images.githubusercontent.com/10143738/149249676-7270bcc0-62aa-41ef-bb8f-7a03b73f1054.png)


Thank you! @glenn-jocher

**Comment by glenn-jocher** [comment]
@VShawn PyTorch models use rectangular inference, so there are less FLOPs involved in the PyTorch inference above than in your ONNX inference. You can see this by looking at your image shapes, i.e. 768x1280 for PyTorch, 1280x1280 for ONNX.

If you want an apples to apples inference example give them square images, i.e. 1280x1280.

You can also look at the --dynamic ONNX export option.
